openssl-dynamic/src/main/c/ssl.c

```diff
@@ -653,20 +633,13 @@ static void ssl_set_thread_id(CRYPTO_THREADID *id)
     CRYPTO_THREADID_set_numeric(id, ssl_thread_id());
 }
 
-static apr_status_t ssl_thread_cleanup(void *data)
+static void ssl_thread_cleanup()
 {
-    UNREFERENCED(data);
     CRYPTO_set_locking_callback(NULL);
     CRYPTO_THREADID_set_callback(NULL);
     CRYPTO_set_dynlock_create_callback(NULL);
```

Comment:
can you clarify? previously a `apr_palloc` was used to allocate the `CRYPTO_dynlock_value` structure. This PR uses `OPENSSL_malloc` instead of the `apr_palloc` method. Either way there is dynamic memory being allocated. I'm not sure where/if it was freed before, but unless you find that OpenSSL is taking ownership in this situation (just setting the variable to `NULL`) then we need to ensure this dynamic memory is cleaned up before we lose the reference to it.